lib/evaluation.py

```diff
@@ -0,0 +1,121 @@
+from sklearn.metrics import *
+from abc import ABC, abstractmethod
+import numpy as np
+
+UNDEFINED = 'Undefined'
+
+
+class Evaluator(ABC):
+	
+	def __init__(self, y_true, y_pred):
+		self.y_true = y_true
+		self.y_pred = y_pred
+	
+	def data_is_valid(self) -> bool:
+		return len(self.y_true) == len(self.y_pred)
+	
+	@abstractmethod
+	def evaluate(self) -> dict:
+		pass
+
+
+class TabularEvaluator(Evaluator):
+	
+	def __init__(self, y_true, y_pred):
+		super().__init__(y_true, y_pred)
+	
+	def evaluate(self) -> dict:
+		pass
+
+
+class RegressionEvaluator(Evaluator):
+	
+	def __init__(self, y_true, y_pred):
+		super().__init__(y_true, y_pred)
+	
+	def evaluate(self) -> dict:
+		if not self.data_is_valid():
+			raise ValueError('Invalid Data')
+		
+		scores = {
+				'rootMeanSquaredError': mean_squared_error(self.y_true, self.y_pred, squared=False),
+				'meanAbsoluteError': mean_absolute_error(self.y_true, self.y_pred),
+				'meanAbsolutePercentageError': mean_absolute_percentage_error(self.y_true, self.y_pred),
+				'rSquared': r2_score(self.y_true, self.y_pred, force_finite=False),
+				'rootMeanSquaredLogError': mean_squared_log_error(self.y_true, self.y_pred),
+		}
+		
+		return scores
+
+
+class ClassificationEvaluator(Evaluator):
+	
+	def __init__(self, y_true, y_prob, thresholds, max_predictions, k=1):
```

Comment:
I thought about maxPredictions, seems like it doesn't make sense in the evaluation API. I plan to remove it from the OpenAPI, what's your opinion?

Reply:
make sense

Reply:
I've removed maxPredictions from OpenAPI, please update your PR as well.